File: octopoints_flutter/lib/db/dao/TeamUserRelationDao.dart
import 'package:floor_annotation/floor_annotation.dart';
import 'package:octopoints_flutter/db/dao/IDao.dart';
import 'package:octopoints_flutter/db/entity/TeamUserRelation.dart';
import 'package:octopoints_flutter/db/entity/UserEntity.dart';

@dao
abstract class TeamUserRelationDao extends IDao<TeamUserRelation> {
  @Query(
      'SELECT U.* FROM users U, teamUserRelation J WHERE J.teamId=:teamId AND J.userId=U.id')
  Future<List<UserEntity>> getTeammates(int teamId);

  @insert
  Future<List<int>> addTeammates(List<TeamUserRelation> list);
}

File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/db/dao/TeamUserRelationDao.kt
package dev.tpavani.octopoints.db.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import dev.tpavani.octopoints.db.entity.TeamUserRelationEntity
import dev.tpavani.octopoints.db.entity.UserEntity

@Dao
interface TeamUserRelationDao : IDao<TeamUserRelationEntity> {
    @Query("SELECT U.* FROM users U, teamUserRelation J WHERE J.teamId=:teamId AND J.userId=U.id")
    suspend fun getTeammates(teamId: Int): List<UserEntity>

    @Insert
    suspend fun addTeammates(list: List<TeamUserRelationEntity>)

    @Query("DELETE FROM teamUserRelation WHERE teamId=:teamId")
    suspend fun deleteTeammates(teamId: Int)
}